src/gateway/bridge/shard_manager.rs

```diff
@@ -96,20 +97,25 @@ use crate::model::gateway::GatewayIntents;
 /// [`Client`]: crate::Client
 #[derive(Debug)]
 pub struct ShardManager {
-    return_value_tx: Sender<Result<(), GatewayError>>,
+    return_value_tx: Mutex<Sender<Result<(), GatewayError>>>,
     /// The shard runners currently managed.
     ///
     /// **Note**: It is highly unrecommended to mutate this yourself unless you need to. Instead
     /// prefer to use methods on this struct that are provided where possible.
     pub runners: Arc<Mutex<HashMap<ShardId, ShardRunnerInfo>>>,
     /// The index of the first shard to initialize, 0-indexed.
-    shard_index: u32,
+    ///
+    /// Atomic integers are used here that way we don't need any mutable references to Self to
+    /// update any of these values.
```

Comment:
```suggestion
    // Atomics are used here to allow for mutation without requiring a mutable reference to self.
```
This should be made a regular comment IMO.

Reply:
I thought I did that, might've accidentally copied and pasted. Thanks!